assets/src/components/form/validations/index.js

```diff
@@ -0,0 +1,194 @@
+		null;
+};
+
+/**
+ * @function
+ * @param {mixed} field
+ * @param {string} fieldName
+ * @return {string|undefined} error string on fail
+ */
+export const matches = ( field, fieldName ) => value => {
+	return ! value || field !== value ?
+		sprintf(
+			__( 'Value should match %s', 'event_espresso' ),
+			fieldName
+		) : null;
+};
+
+/**
+ * @function
+ * @param {number} max
+ * @return {string|null} error string on fail
+ */
+export const maxFloat = max => value => {
+	max = parseFloat( max, 10 );
+	return parseFloat( value, 10 ) > max ?
+		sprintf(
+			_n(
+				'Number should be less than %f',
+				'Number should be less than %f',
+				max,
+				'event_espresso'
+			),
+			max
+		) : null;
+};
+
+/**
+ * @function
+ * @param {number} max
+ * @return {string|null} error string on fail
+ */
+export const maxLength = ( max = 1024 ) => value => {
+	max = parseInt( max, 10 );
+	return value !== undefined && parseInt( value.length, 10 ) > max ?
+		sprintf(
+			_n(
+				'Value should not exceed %d character in length',
+				'Value should not exceed %d characters in length',
+				max,
+				'event_espresso'
+			),
+			max
+		) : null;
+};
+
+/**
+ * @function
+ * @param {number} max
+ * @return {string|null} error string on fail
+ */
+export const maxNumber = max => value => {
+	max = parseInt( max, 10 );
+	return parseInt( value, 10 ) > max ?
+		sprintf(
+			_n(
+				'Number should be less than %d',
+				'Number should be less than %d',
+				max,
+				'event_espresso'
+			),
+			max
+		) : null;
+};
+
+/**
+ * @function
+ * @param {number} min
+ * @return {string|null} error string on fail
+ */
+export const minFloat = min => value => {
+	min = parseFloat( min, 10 );
+	return parseFloat( value, 10 ) < min ?
+		sprintf(
+			_n(
+				'Number should be greater than %f',
+				'Number should be greater than %f',
+				min,
+				'event_espresso'
+			),
+			min
+		) : null;
+};
+
+/**
+ * @function
+ * @param {number} min
+ * @return {string|null} error string on fail
+ */
+export const minLength = ( min = 1 ) => value => {
+	min = parseInt( min, 10 );
+	return value === undefined ||
+	( value && parseInt( value.length, 10 ) < min ) ?
+		sprintf(
+			_n(
+				'Value should be at least %d character in length',
+				'Value should be at least %d characters in length',
+				min,
+				'event_espresso'
+			),
+			min
+		) : null;
+};
+
+/**
+ * @function
+ * @param {number} min
+ * @return {string|null} error string on fail
+ */
+export const minNumber = min => value => {
+	min = parseInt( min, 10 );
+	return parseInt( value, 10 ) < min ?
+		sprintf(
+			_n(
+				'Number should be greater than %d',
+				'Number should be greater than %d',
+				min,
+				'event_espresso'
+			),
+			min
+		) : null;
+};
+
+/**
+ * @function
+ * @param {mixed} value
+ * @return {string|null} error string on fail
+ */
+export const required = value => {
+	return ! value ?
+		__( 'This is a required field', 'event_espresso' ) :
+		null;
```

Comment:
not sure this really fits the description of "validators"?  Can you give an example of how this would be implemented?

Reply:
if you add this validator to an input, then the form can not be submitted without entering a value for that input. So a user would see an error under the empty input saying:

> This is a required field

which given my comment above, should be changed to something more user friendly